cobalt/dom/lottie_player.cc

```diff
@@ -298,9 +299,28 @@ void LottiePlayer::UpdateAnimationData() {
       return;
     }
 
+    auto isBase64Encoded = [](const std::string& url) -> bool {
+      size_t pos = url.find(',');
+      if (pos != std::string::npos) {
+        std::string beforeComma = url.substr(0, pos);
+        return beforeComma.find("base64") != std::string::npos;
+      }
+      return false;
+    };
+
+    // Force base64 encoding of plain/raw inputs, to ensure no URL escaping
+    // occurs.
+    GURL resource_url = selected_source;
+    if (selected_source.SchemeIs("data") && !isBase64Encoded(src)) {
+      auto encoded = ForgivingBase64Encode(src.substr(src.find(',') + 1));
```

Comment:
nit: I personally find it hard to read code that uses `auto` then dereferences it

Reply:
Changed the return type to optional, it's a lot more verbose.

Reply:
If this is an optional and you don't like the verbosity you can communicate what it is by checking `encoded.has_value()` and `encoded.value()` in place of of `encoded`/`*encoded`